Sources/Sentry/SentryFileManager.m

```diff
@@ -669,19 +690,79 @@ - (void)createPathsWithOptions:(SentryOptions *_Nonnull)options
     self.envelopesPath = [self.sentryPath stringByAppendingPathComponent:EnvelopesPathComponent];
 }
 
-- (BOOL)createDirectoryIfNotExists:(NSString *)path error:(NSError **)error
+#if SENTRY_TARGET_PROFILING_SUPPORTED
+/**
+ * @note This method must be statically accessible because it will be called during app launch,
+ * before any instance of @c SentryFileManager exists, and so wouldn't be able to access this path
+ * from an objc property on it like the other paths.
+ */
+NSString *
+sentryApplicationSupportPath(void)
 {
-    if (![[NSFileManager defaultManager] createDirectoryAtPath:path
-                                   withIntermediateDirectories:YES
-                                                    attributes:nil
-                                                         error:error]) {
-        *error = NSErrorFromSentryErrorWithUnderlyingError(kSentryErrorFileIO,
-            [NSString stringWithFormat:@"Failed to create the directory at path %@.", path],
-            *error);
-        return NO;
+    static NSString *sentryApplicationSupportPath;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        NSArray *paths = NSSearchPathForDirectoriesInDomains(
+            NSApplicationSupportDirectory, NSUserDomainMask, YES);
+        NSString *applicationSupportDirectory = [paths firstObject];
+        sentryApplicationSupportPath =
+            [applicationSupportDirectory stringByAppendingPathComponent:@"io.sentry"];
+    });
+    return sentryApplicationSupportPath;
+}
+
+NSURL *
+launchProfileConfigFileURL(void)
+{
+    static NSURL *sentryLaunchConfigFileURL;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        sentryLaunchConfigFileURL =
+            [NSURL fileURLWithPath:[sentryApplicationSupportPath()
+                                       stringByAppendingPathComponent:@"profileLaunch"]];
+    });
+    return sentryLaunchConfigFileURL;
+}
+
+NSDictionary<NSString *, NSNumber *> *_Nullable appLaunchProfileConfiguration(void)
+{
+    NSError *error;
+    NSDictionary<NSString *, NSNumber *> *config = [NSDictionary<NSString *, NSNumber *>
+        dictionaryWithContentsOfURL:launchProfileConfigFileURL()
+                              error:&error];
+    SENTRY_CASSERT(
+        error == nil, @"Encountered error trying to retrieve app launch profile config: %@", error);
```

Comment:
yes, this is one of the things i mentioned to you the other day in DM that i'd encountered, i already have a fix for it locally.

Reply:
to provide more context, there was actually another issue at play as well. if a launch profile started, but the new call to SentrySDK.startWithOptions would've done something that'd disable launch profiling, like setting sample rates to 0, then it would remove this file, and then after that we'd try to start the tx, which would go looking for it.

i solved this by moving a config to a backup location before making any changes, so there's a little bit more logic going into SentryFileManager to manage all that.

Reply:
This is now outdated as we no longer use the backup-config approach.